File: CoolkyDBTools/Recipe.cs
using System;
using System.Collections.Generic;
using Realms;

namespace CoolkyTools
{
    public class Recipe : RealmObject
    {
        public string Id { get; set; }
        public string DishName { get; set; }
        public string CookTime { get; set; }
        public string Cuisine { get; set; }
        public string Type { get;  }
        public string PortionAmount { get;  }
        public IList<string> Ingredients { get; }
        public IList<string> Steps { get; }

        public Recipe() { }

        public Recipe(string id, string dishName, string cookTime, string cuisine, string type, string portionAmount,
                List<string> ingredients, List<string> steps)
        {
            Id = id;
            DishName = dishName;
            CookTime = cookTime;
            Cuisine = cuisine;
            Type = type;
            Ingredients = ingredients;
            Steps = steps;
            PortionAmount = portionAmount;
        }

        public void Print()
        {
            Console.WriteLine(Id);
            Console.WriteLine(DishName);
            Console.WriteLine(Type);
            Console.WriteLine(Cuisine);
            Console.WriteLine(CookTime);
            Console.WriteLine(PortionAmount);

            foreach (var ingredient in Ingredients)
            {
                Console.WriteLine(ingredient);
            }

            foreach (var step in Steps)
            {
                Console.WriteLine(step);
            }

            Console.WriteLine();
        }
    }
}

File: Coolky/app/src/main/java/com/example/coolky/database/models/Recipe.kt
package com.example.coolky.database.models

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Recipe : RealmObject()
{
    @PrimaryKey
    public var Id: String? = null

    public var DishName: String? = null
    public var CookTime: Int = 0
    public var Cuisine: String? = null
    public var Type: String? = null
    public var PortionAmount: Int = 0
    public var PictureUrl: String? = null
    public var WebSite: String? = null
    public var Steps: RealmList<String> = RealmList()
}